src/extension/bitcoin.rs

```diff
@@ -189,35 +190,61 @@ impl extension::Jet for BtcNode {
+                Cmr::tag_iv(b"Simplicity-Draft\x1fPrimitive\x1fBitcoin\x1ftotalInputValue")
+            }
             BtcNode::CurrentPrevOutpoint => {
-                Cmr::new(b"SimplicityPrimitiveBitcoinx1fcurrentPrevOutpoint")
+                Cmr::tag_iv(b"Simplicity-Draft\x1fPrimitive\x1fBitcoin\x1fcurrentPrevOutpoint")
+            }
+            BtcNode::CurrentValue => {
+                Cmr::tag_iv(b"Simplicity-Draft\x1fPrimitive\x1fBitcoin\x1fcurrentValue")
+            }
+            BtcNode::CurrentSequence => {
+                Cmr::tag_iv(b"Simplicity-Draft\x1fPrimitive\x1fBitcoin\x1fcurrentSequence")
+            }
+            BtcNode::CurrentIndex => {
+                Cmr::tag_iv(b"Simplicity-Draft\x1fPrimitive\x1fBitcoin\x1fcurrentIndex")
             }
-            BtcNode::CurrentValue => Cmr::new(b"SimplicityPrimitiveBitcoinx1fcurrentValue"),
-            BtcNode::CurrentSequence => Cmr::new(b"SimplicityPrimitiveBitcoinx1fcurrentSequence"),
-            BtcNode::CurrentIndex => Cmr::new(b"SimplicityPrimitiveBitcoinx1fcurrentIndex"),
             BtcNode::InputPrevOutpoint => {
-                Cmr::new(b"SimplicityPrimitiveBitcoinx1finputPrevOutpoint")
+                Cmr::tag_iv(b"Simplicity-Draft\x1fPrimitive\x1fBitcoin\x1finputPrevOutpoint")
+            }
+            BtcNode::InputValue => {
+                Cmr::tag_iv(b"Simplicity-Draft\x1fPrimitive\x1fBitcoin\x1finputValue")
+            }
+            BtcNode::InputSequence => {
+                Cmr::tag_iv(b"Simplicity-Draft\x1fPrimitive\x1fBitcoin\x1finputSequence")
+            }
+            BtcNode::NumOutputs => {
+                Cmr::tag_iv(b"Simplicity-Draft\x1fPrimitive\x1fBitcoin\x1fnumOutputs")
+            }
+            BtcNode::TotalOutputValue => {
+                Cmr::tag_iv(b"Simplicity-Draft\x1fPrimitive\x1fBitcoin\x1ftotalOutputValue")
+            }
+            BtcNode::OutputValue => {
+                Cmr::tag_iv(b"Simplicity-Draft\x1fPrimitive\x1fBitcoin\x1foutputValue")
+            }
+            BtcNode::OutputScriptHash => {
+                Cmr::tag_iv(b"Simplicity-Draft\x1fPrimitive\x1fBitcoin\x1foutputScriptHash")
+            }
+            BtcNode::ScriptCMR => {
+                Cmr::tag_iv(b"Simplicity-Draft\x1fPrimitive\x1fBitcoin\x1fscriptCMR")
             }
-            BtcNode::InputValue => Cmr::new(b"SimplicityPrimitiveBitcoinx1finputValue"),
-            BtcNode::InputSequence => Cmr::new(b"SimplicityPrimitiveBitcoinx1finputSequence"),
-            BtcNode::NumOutputs => Cmr::new(b"SimplicityPrimitiveBitcoinx1fnumOutputs"),
-            BtcNode::TotalOutputValue => Cmr::new(b"SimplicityPrimitiveBitcoinx1ftotalOutputValue"),
-            BtcNode::OutputValue => Cmr::new(b"SimplicityPrimitiveBitcoinx1foutputValue"),
-            BtcNode::OutputScriptHash => Cmr::new(b"SimplicityPrimitiveBitcoinx1foutputScriptHash"),
-            BtcNode::ScriptCMR => Cmr::new(b"SimplicityPrimitiveBitcoinx1fscriptCMR"),
```

Comment:
At the moment, the main method of this crate fails because the CMRs of Simplicity-Bitcoin programs are different. I added `\x1f` where it was missing. This seems to be in agreement with the whitepaper. Should I simply update the CMRs that are hard-coded in `main.rs`?

Reply:
Yep, that sounds like the right thing to do.

Reply:
Example Schnorr1 is no longer in the C implementation of Simplicity. Instead, we have [Schnorr0](https://github.com/ElementsProject/simplicity/blob/master/C/schnorr0.c) and [Schnorr6](https://github.com/ElementsProject/simplicity/blob/master/C/schnorr6.c). Of course, I could copy the CMRs that Rust produces and fix the assertions, but this doesn't seem right. If we replace the current Schnorr1 with Schnorr0 or Schnorr6, then we can use "official" CMRs from C. As for the Fibonacci program, there doesn't seem to be any trace of it in C. I think, [I found it in the Haskell implementation](https://github.com/ElementsProject/simplicity/blob/master/Haskell/Tests/Simplicity/Programs/Example.hs), but there is no hard-coded CMR. Maybe, @sanket1729 has ideas on how to test the Rust CMRs against C.